### PR TITLE
Update developer docs

### PR DESCRIPTION
**Pull request by Hofer-Julian**
- Update and extend developer docs
- Remove instructions on how to install Ribasim as Julia package
- Update run/README.md

### CODE REVIEW

File: run/README.md [hunk 1]
```diff
@@ -1,13 +1,4 @@
-This folder is a Julia project meant for running Ribasim simulations and post-processing.
+This folder is a Julia project meant for running Ribasim simulations and do post-processing.
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply run/README.md:1]: You are right, I've reverted this change

File: docs/developer.qmd [hunk 2]
```diff
@@ -2,40 +2,111 @@
...
+
+## Setup Revise.jl
+
+Revise.jl is a library that allows you to modify code and use the changes without restarting Julia. 
+You can let it start automatically by following these [instructions](https://timholy.github.io/Revise.jl/stable/config/#Using-Revise-by-default-1).
+
+## Install Quarto
+
+Quarto is the publishing system that builds our documentation.
+The installation instructions can be found on their [web page](https://quarto.org/docs/download/).
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply docs/developer.qmd:30]: Sure.

File: docs/developer.qmd [hunk 3]
```diff
@@ -2,40 +2,111 @@
...
+
+The documentation also includes [Jupyter notebooks](https://jupyter.org/).
+To edit and execute them run the following commands in your Julia REPL.
+The first time, it will offer to download Conda and Jupyter for you.
+Say yes, and a browser tab will open with Jupyter loaded.
+
+```julia
+julia> using IJulia
+julia> jupyterlab()
+```
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply docs/developer.qmd:89]: >  Should we mention that as an alternative?

Yes we should, good point.
For full quarto support, we still need Jupyter itself.
I've noticed that vscode doesn't support creating raw cells, even worse it converts existing raw cells into code cells.
This breaks this functionality: https://quarto.org/docs/tools/jupyter-lab.html#yaml-front-matter

File: docs/developer.qmd [hunk 4]
```diff
@@ -2,40 +2,111 @@
...
+
+```julia
+julia> using Ribasim
+```
+
+Then open the Pkg REPL by typing `]` and execute:
+
+```julia
+pkg> test
+```
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply docs/developer.qmd]: Done.

File: docs/developer.qmd [hunk 5]
```diff
@@ -2,40 +2,121 @@
...
+
+The documentation also includes [Jupyter notebooks](https://jupyter.org/).
+To edit and execute them run the following commands in your Julia REPL.
+The first time, it will offer to download Conda and Jupyter for you.
+Say yes, and a browser tab will open with Jupyter loaded.
+
+```julia
+julia> using IJulia
+julia> jupyterlab(".")
```
Comment (visr) [inline docs/developer.qmd:94]:
```suggestion
julia> jupyterlab(dir=".")
```

It's a keyword argument, and unlike Python providing it as a positional one doesn't work.

**Review by visr — APPROVED** [review]
Thanks! Two minor last suggestions, with that this is good to go.

File: docs/developer.qmd (hunk 5 above)
Comment (Hofer-Julian) [reply docs/developer.qmd:94]: Oh good catch, I should have tried the command first.

**Pull request merged**